File: solutions/2026-06-17/solution.py
# Reverse Linked List
# Difficulty: Easy
# URL: https://leetcode.com/problems/reverse-linked-list/

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

def reverseList(self, head: ListNode) -> ListNode:
    if not head or not head.next:
        return head

    new_head = head
    while head and head.next:
        next_node = head.next
        head.next = next_node.next
        next_node.next = new_head
        new_head = next_node
        head = head.next
    return new_head

File: solutions/2026-06-17/solution.java
// Reverse Linked List
// Difficulty: Easy
// URL: https://leetcode.com/problems/reverse-linked-list/

class ListNode {
    int val;
    ListNode next;

    ListNode() {}
    ListNode(int val) { this.val = val; }
    ListNode(int val, ListNode next) { this.val = val; this.next = next; }

    public ListNode reverseList(ListNode head) {
        if (head == null || head.next == null) {
            return head;
        }

        ListNode newHead = head;
        while (head != null && head.next != null) {
            ListNode nextNode = head.next;
            head.next = nextNode.next;
            nextNode.next = newHead;
            newHead = nextNode;
            head = head.next;
        }
        return newHead;
    }
}